src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -23,13 +31,37 @@ namespace Microsoft.VisualStudio.LanguageServerClient.Razor
     [ContentType(RazorLSPConstants.RazorLSPContentTypeName)]
     internal class RazorLSPTextViewConnectionListener : ITextViewConnectionListener
     {
+        private const string RazorEditorTrackedView = "RazorEditorTrackedView";
```

Comment:
Instead of storing 3 different properties I'd suggest making a singular RazorEditorOptionsTracker record / type to encapsulate all the important information. That way you can extract/update a single piece of information, should make all the state management easier to reason about.